ibis/backends/duckdb/__init__.py

```diff
@@ -172,7 +192,17 @@ def register(
         ir.Table
             The just-registered table
         """
-        if isinstance(source, (str, Path)):
+        if isinstance(source, str) and source.startswith("s3://"):
+            sql, (table_name, dataset) = _generate_view_code(
+                source, table_name=table_name
+            )
+            with self._safe_raw_sql(sql) as cursor:
+                # DuckDB normally auto-detects Arrow Datasets that are defined
+                # in local variables but the `dataset` variable won't be local
+                # by the time we execute against this so we register it
+                # explicitly.
+                cursor.cursor.c.register(table_name, dataset)
```

Comment:
Should this be `cursor.cursor.register`? I vaguely recall that there's some reason related to connection pooling to prefer using the proxy, but if this works then it's probably fine for now.